### PR TITLE
Added additional methods for storage for the cases when the developer need to work with `StorageAsRef` and `StorageAsMut`

### PR DESCRIPTION
**Pull request by xgreenx**

### CODE REVIEW

File: fuel-storage/src/lib.rs [hunk 1]
```diff
@@ -123,6 +123,14 @@ pub struct StorageRef<'a, T: 'a + ?Sized, Type: Mappable>(&'a T, core::marker::P
 pub trait StorageAsRef {
     #[inline(always)]
     fn storage<Type>(&self) -> StorageRef<Self, Type>
+    where
+        Type: Mappable,
```
Comment (ControlCplusControlV) [inline fuel-storage/src/lib.rs:127]: What is the main motivation behind this PR? Is it blocking you somewhere else? If so do we still have time to sneak it in before next release for everything?

File: fuel-storage/src/impls.rs [hunk 2]
```diff
@@ -53,6 +53,16 @@ impl<'a, T: StorageInspect<Type>, Type: Mappable> StorageRef<'a, T, Type> {
     }
 }
 
+impl<'a, T, Type: Mappable> StorageRef<'a, T, Type> {
+    #[inline(always)]
+    pub fn root<Key>(self, key: &Key) -> Result<MerkleRoot, T::Error>
```
Comment (freesig) [inline fuel-storage/src/impls.rs:58]: Is this a new rust syntax? Where is the `for` in this `impl`

**Review by freesig — APPROVED** [review]
Makes sense

**Review by ControlCplusControlV — APPROVED** [review]

File: fuel-storage/src/lib.rs (hunk 1 above)
Comment (xgreenx) [reply fuel-storage/src/lib.rs:127]: It it to improve some part of the code in the future to avoid `(&mut database)` or `(&**database)`

**Pull request merged**